Marlin/src/HAL/HAL_DUE/usb/conf_access.h

```diff
@@ -48,7 +48,8 @@
 #define _CONF_ACCESS_H_
 
 #include "compiler.h"
-
+#include "../../../core/macros.h"
+#include "../../../../Configuration.h"
```

Comment:
Can't use `MarlinConfig.h` because `conf_access.h` is included from a `.c` file and `MarlinConfig.h` will eventually try to include a file with a `class` in it.

Reply:
You are right. If you need the SDSUPPORT config option, then you have to include those files. I completely agree with @teemuatlut

Reply:
Hmm. I guess we can't easily instruct the compiler to treat `.c` files as `.cpp` files either.

Reply:
There is a way to force compilation as cpp. It could even be possible to rename c files to cpp. I didn't want to alter ASF sources so much...
If you are trying to force all your .c files to be compiled using the
c++ compiler, you can use the switch "-x c++" to force the compilation
language, as noted in the gcc manual:
http://gcc.gnu.org/onlinedocs/gcc-4.1.1/gcc/Overall-Options.html#Over..

Reply:
Hmm. Unfortunately unless it can be set from within the project itself it's not going to be helpful for Arduino IDE users. The only way we can manage it is to rename `.c` to `.cpp`.

See if including `MarlinConfigPre.h` instead of `MarlinConfig.h` allows it to work.